satpy/writers/cf_writer.py

```diff
@@ -514,13 +515,28 @@ def _collect_datasets(self, datasets, epoch=EPOCH, flatten_attrs=False, exclude_
 
         return datas, start_times, end_times
 
-    def update_encoding(self, dataset, to_netcdf_kwargs):
+    @staticmethod
+    def update_encoding(dataset, to_netcdf_kwargs):
         """Update encoding.
 
-        Avoid _FillValue attribute being added to coordinate variables (https://github.com/pydata/xarray/issues/1865).
+        Preserve chunk sizes, avoid fill values in coordinate variables and make sure that
+        time & time bounds have the same units.
         """
         other_to_netcdf_kwargs = to_netcdf_kwargs.copy()
         encoding = other_to_netcdf_kwargs.pop('encoding', {}).copy()
+
+        # If not specified otherwise by the user, preserve current chunks.
+        for var_name, variable in dataset.variables.items():
+            if variable.chunks:
+                chunks = tuple(
+                    np.stack([variable.data.chunksize,
+                              variable.shape]).min(axis=0)
+                )  # Chunksize may not exceed shape
+                encoding.setdefault(var_name, {})
+                encoding[var_name].setdefault('chunksizes', chunks)
```

Comment:
Can you extract this at a separate method for readability?

Reply:
Sure, will do!